node/pkg/watchers/solana/client.go

```diff
@@ -398,6 +413,20 @@ func (s *SolanaWatcher) Run(ctx context.Context) error {
 		}
 	})
 
+	if s.commitment == rpc.CommitmentType("finalized") {
+		s.ccqLogger = logger.With(zap.String("component", "ccqsol"))
+		common.RunWithScissors(ctx, s.errC, "solana_fetch_query_req", func(ctx context.Context) error {
```

Comment:
Nice job wrapping this since the handler can crash if a chain id security invariant is false.